oblib/data_model.py

```diff
@@ -460,7 +461,15 @@ def __init__(self, concept, context, unit, value, decimals=2):
         self.value = value
         self.context = context
         self.unit = unit
+
+        if decimals is not None and precision is not None:
+            raise OBException("Fact given both precision and decimals - use only one.")
         self.decimals = decimals
+        self.precision = precision
+        if decimals is None and precision is None:
```

Comment:
I think this is confusing.  The XRBL spec Sect 4.6.3 says 1) for numeric item types, either Decimal or Precision (not neither and not both), and 2) for non-numeric types, neither Decimal nor Precision.

What about making two constant lists, e.g., NUMERIC_TYPES = ['decimalItemType', …] and NON_NUMERIC_TYPES = ['stringItemType', …] and checking membership to apply the rules for assigning Decimal and Precision?  I'm sure that's not the best implementation but hope it conveys the idea.

Reply:
I would probably create a map with 'decimalItemType' and 'stringItemType' being keys and boolean True to indicate Numeric and False to indicate non-Numeric.

Reply:
If it's a non-numeric type, it won't have a Unit, and therefore any decimals/precision property the Fact has does not get written out in Fact._toJSON() or Fact._toXML() unless it's numeric.

Presence/absence of Unit is not a foolproof way to determine whether it's numeric or not, I know. The approach of mapping every itemType to be numeric or non-numeric is going to be problematic as well, however, because there are so many itemTypes and I'm not sure there's an exhaustive list everywhere. I keep discovering new ones...

Reply:
How about this: I'll remove the default for now. It'll be up to the user to decide whether they need to include a decimal/precision or not.

Then I'll file a ticket for a future version to say that we should detect the difference between numeric and non-numeric types, and for numeric types either require a decimals/precision or offer a default decimals/precision. Sound good?

Reply:
OK, i've removed the code that sets default decimal, and filed ticket #131 https://github.com/SunSpecOrangeButton/pyoblib/issues/131

Reply:
Sounds good.